src/Microsoft.Extensions.SecretManager.Tools/Internal/MsBuildProjectContextFactory.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.ProjectModel.Internal;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    public class MsBuildProjectContextFactory
+    {
+        private readonly DotNetCoreSdkResolver _sdkResolver;
+
+        public MsBuildProjectContextFactory()
+        {
+            var dotNetHome = Path.GetDirectoryName(new Muxer().MuxerPath);
```

Comment:
In https://github.com/aspnet/DotNetTools/pull/180 I have moved this into an abstraction that represents `MsBuildContext`